### PR TITLE
fix(haxe) fix new keyword false positives

### LINKED ISSUE #3992: (Haxe) using `new` at the start of a variable name highlights it separately from the rest of the variable name
**Describe the issue**

When naming a variable with `new` at the start of the variable name (ex: `newAngle`), it highlights `new` separately from the rest of the variable (highlights `new` one color, then highlights `Angle` another color)

**Which language seems to have the issue?**
[Haxe](https://haxe.org)

**Are you using `highlight` or `highlightAuto`?**

I believe I am using `highlight` / `highlightAll`

**Sample Code to Reproduce**
(Sample code is from the [haxeflixel.com documentation](https://haxeflixel.com/documentation/groundwork/), the hosted documentation is using a bit of custom highlighting, but it's going to move to highlightjs, which you can test on [dev.haxeflixel.com](https://dev.haxeflixel.com/documentation/groundwork/))
```hx
var newAngle:Float = 0;
 if (up)
 {
     newAngle = -90;
     if (left)
         newAngle -= 45;
     else if (right)
         newAngle += 45;
 }
 else if (down)
 {
     newAngle = 90;
     if (left)
         newAngle += 45;
     else if (right)
         newAngle -= 45;
 }
 else if (left)
     newAngle = 180;
 else if (right)
     newAngle = 0;
```

**Expected behavior**

Usually we'd want `new` to be highlighted differently (ex: creating a new variable `state = new FlxState()`), but when it's used as a variable, it should be highlighted as a variable rather than whatever the current implementation is for the `new` keyword.

**Additional context**
This error can be easily tested from the highlightjs online demo with nearly any theme: https://highlightjs.org/demo

### PR DESCRIPTION
**Pull request by ninjamuffin99**
Changes the regex in the haxe.js highlighting to only highlight `new` as a keyword if it's not used within a variable/class name, using `beginKeywords: 'new'` instead of `new *`

Resolves #3992 
### Changes

Chnages the regex to use `beginKeywords: 'new'` instead of *, so that it can capture `new` as a keyword, but not when it's used within a variable name
### Checklist
- [x] Added markup tests
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by ninjamuffin99** [comment]
I will look into the markup tests sometime soon / hopefully later today

File: src/languages/haxe.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ export default function(hljs) {
       },
       {
         className: 'type', // instantiation
-        begin: /new */,
+        begin: /\bnew\b/,
```
Comment (joshgoebel) [inline src/languages/haxe.js:92]: Would this also work:

```
beginKeywords: 'new',
```

Comment (joshgoebel) [reply src/languages/haxe.js:92]: Depending on what the goal is here a multi-matcher might also be useful... see the docs on `match` for an example... also some of the newer grammers like Ruby or Wren have examples of this in action.  The Haxe grammar looks to be quite old and dusty.

Let me know if you have any questions.

Comment (ninjamuffin99) [reply src/languages/haxe.js]: I've updated it to use beginKeywords and works good 🤝

**Comment by ninjamuffin99** [comment]
added markup tests, and changed to use `beginKeywords: new`

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**